sematic/resolvers/resource_managers/server_manager.py

```diff
@@ -2,21 +2,32 @@
 import logging
 import time
 from threading import Thread
-from typing import List, Set
+from typing import List, Optional, Set
 
 # Sematic
 from sematic import api_client
+from sematic.config.server_settings import ServerSettingsVar
 from sematic.plugins.abstract_external_resource import AbstractExternalResource
 from sematic.resolvers.abstract_resource_manager import AbstractResourceManager
 
 logger = logging.getLogger(__name__)
 
 
+_DEFAULT_INTERVAL_BETWEEN_UPDATES_SECONDS = 60
+
+
 class ServerResourceManager(AbstractResourceManager):
     """ResourceManager which uses server APIs to manage external resource metadata."""
 
-    def __init__(self, update_poll_interval_seconds: int = 600) -> None:
+    def __init__(self, update_poll_interval_seconds: Optional[int] = None) -> None:
         super().__init__()
+        if update_poll_interval_seconds is None:
+            update_poll_interval_seconds_str = api_client.get_server_env_settings().get(
+                ServerSettingsVar.SEMATIC_RESOURCE_UPDATE_POLL_INTERVAL_SECONDS.value,
+                str(_DEFAULT_INTERVAL_BETWEEN_UPDATES_SECONDS),
```

Comment:
I believe the need to have this call here, together with the fact that the polling itself is done by the Resolver, suggest this configuration should be a User one.

Reply:
Here are my reasons for thinking it should be a server setting:

- users have an incentive to make this config as short as possible, but it is the server maintainer who will want it to be changed if load starts disrupting user jobs
- ideally we want all users to poll at the same rate. That's not enforceable if each user sets it independently
- ideally a given pipeline should use the same polling rate whether executed by user A or user B. That would not be the behavior if this was a user setting

WDYT?